src/module/BaseKernelRateProvider.sol

```diff
@@ -27,11 +27,11 @@ abstract contract BaseKernelRateProvider is IKernelProvider {
     function tryGetVaultAsset(address vault) public view returns (address) {
         try IKernelVault(vault).getAsset() returns (address asset) {
             if (IStakerGateway(getStakerGateway()).getVault(asset) != vault) {
-                return address(0);
+                revert AssetMismatch(vault, asset);
             }
             return asset;
         } catch {
-            return address(0);
```

Comment:
let's return address(0) on this branch. 

if we revert here the whole getRate() reverts and its a show stopper, you have to then try catch, in the getRate function to have any code after.

This function already does that for you.

Reply:
Are you suggesting we throw the `AssetMismatch` error but NOT the second `ExternalCallFailed` error?